bodhi/tests/server/consumers/test_updates.py

```diff
@@ -117,6 +117,70 @@ def test_request_testing(self, work_on_bugs, fetch_test_cases):
         self.assertTrue(isinstance(fetch_test_cases.mock_calls[0][1][1],
                                    sqlalchemy.orm.session.Session))
 
+    @mock.patch('bodhi.server.buildsys.DevBuildsys.getTaskRequest', side_effect=IOError("oh no"))
+    def test_scm_url_fail(self, getTaskRequest):
+        """Test that the scm_url is added when communication with Koji fails."""
+        hub = mock.MagicMock()
+        hub.config = {'environment': 'environment',
+                      'topic_prefix': 'topic_prefix'}
+        h = updates.UpdatesHandler(hub)
+        h.db_factory = base.TransactionalSessionMaker(self.Session)
+        # Throw a bogus bug id in there to ensure it doesn't raise AssertionError.
+        message = {
+            'topic': 'bodhi.update.request.testing',
+            'body': {'msg': {'update': {'alias': u'bodhi-2.0-1.fc17'}}}}
+
+        h.consume(message)
+
+        b = models.Build.query.filter(models.Build.nvr == u'bodhi-2.0-1.fc17').one()
+        # Since we could not communicate with Koji, the scm_url should still be None.
+        self.assertEqual(b.scm_url, None)
+        getTaskRequest.assert_called_once_with(127621)
+
+    @mock.patch('bodhi.server.models.Build.get_scm_url')
+    def test_scm_url_present(self, get_scm_url):
+        """Test that the scm_url is left alone when it's already set in the database."""
+        b = models.Build.query.filter(models.Build.nvr == u'bodhi-2.0-1.fc17').one()
+        b.scm_url = u'git://some.cool.host/rpms/neat#master'
+        self.db.commit()
+        hub = mock.MagicMock()
+        hub.config = {'environment': 'environment',
+                      'topic_prefix': 'topic_prefix'}
+        h = updates.UpdatesHandler(hub)
+        h.db_factory = base.TransactionalSessionMaker(self.Session)
+        # Throw a bogus bug id in there to ensure it doesn't raise AssertionError.
```

Comment:
@jeremycline you caught me in a copypaste crime! I'm applying this patch to fix it:

```
diff --git a/bodhi/tests/server/consumers/test_updates.py b/bodhi/tests/server/consumers/test_updates.py
index 99db898..abacc7a 100644
--- a/bodhi/tests/server/consumers/test_updates.py
+++ b/bodhi/tests/server/consumers/test_updates.py
@@ -125,7 +125,6 @@ class TestUpdatesHandlerConsume(base.BaseTestCase):
                       'topic_prefix': 'topic_prefix'}
         h = updates.UpdatesHandler(hub)
         h.db_factory = base.TransactionalSessionMaker(self.Session)
-        # Throw a bogus bug id in there to ensure it doesn't raise AssertionError.
         message = {
             'topic': 'bodhi.update.request.testing',
             'body': {'msg': {'update': {'alias': u'bodhi-2.0-1.fc17'}}}}
@@ -148,7 +147,6 @@ class TestUpdatesHandlerConsume(base.BaseTestCase):
                       'topic_prefix': 'topic_prefix'}
         h = updates.UpdatesHandler(hub)
         h.db_factory = base.TransactionalSessionMaker(self.Session)
-        # Throw a bogus bug id in there to ensure it doesn't raise AssertionError.
         message = {
             'topic': 'bodhi.update.request.testing',
             'body': {'msg': {'update': {'alias': u'bodhi-2.0-1.fc17'}}}}
@@ -168,7 +166,6 @@ class TestUpdatesHandlerConsume(base.BaseTestCase):
                       'topic_prefix': 'topic_prefix'}
         h = updates.UpdatesHandler(hub)
         h.db_factory = base.TransactionalSessionMaker(self.Session)
-        # Throw a bogus bug id in there to ensure it doesn't raise AssertionError.
         message = {
             'topic': 'bodhi.update.request.testing',
             'body': {'msg': {'update': {'alias': u'bodhi-2.0-1.fc17'}}}}
```